src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Field.cs

```diff
@@ -12,11 +11,65 @@ public override BoundNode VisitFieldAccess(BoundFieldAccess node)
         {
             BoundExpression rewrittenReceiver = VisitExpression(node.ReceiverOpt);
 
-            // field access of a tuple actually accesses the underlying field.
-            FieldSymbol field = (node.FieldSymbol as TupleFieldSymbol)?.UnderlyingTypleFieldSymbol ??
-                                node.FieldSymbol;
+            var tupleField = node.FieldSymbol as TupleFieldSymbol;
+            if (tupleField != null)
+            {
+                return RewriteTupleFieldAccess(node, rewrittenReceiver);
+            }
+            else
+            {
+                return MakeFieldAccess(node.Syntax, rewrittenReceiver, node.FieldSymbol, node.ConstantValue, node.ResultKind, node.Type, node);
+            }
+        }
+
+        /// <summary>
+        /// Converts access to a tuple instance into access into the underlying ValueTuple(s).
+        ///
+        /// For instance, tuple.Item8
+        /// produces fieldAccess(field=Item1, receiver=fieldAccess(field=Rest, receiver=ValueTuple for tuple))
+        /// </summary>
+        private BoundExpression RewriteTupleFieldAccess(BoundFieldAccess node, BoundExpression rewrittenReceiver)
+        {
+            var tupleField = (TupleFieldSymbol)node.FieldSymbol;
+            var tupleType = (TupleTypeSymbol)tupleField.ContainingSymbol;
+
+            int fieldRemainder;
+            int loop = TupleTypeSymbol.NumberOfValueTuples(tupleField.Position, out fieldRemainder);
+
+            NamedTypeSymbol currentLinkType = tupleType.UnderlyingTupleType;
+
+            if (loop > 1)
+            {
+                WellKnownMember wellKnownTupleRest = TupleTypeSymbol.GetTupleTypeMember(TupleTypeSymbol.Chunk8, TupleTypeSymbol.Chunk8);
+                var tupleRestField = (FieldSymbol)Binder.GetWellKnownTypeMember(_compilation, wellKnownTupleRest, _diagnostics, syntax: node.Syntax);
+                if ((object)tupleRestField == null)
+                {
+                    return node;
+                }
+
+                // make nested field accesses to Rest
+                do
+                {
+                    FieldSymbol nestedFieldSymbol = tupleRestField.AsMember(currentLinkType);
+
+                    currentLinkType = (NamedTypeSymbol)currentLinkType.TypeArgumentsNoUseSiteDiagnostics[TupleTypeSymbol.Chunk7];
```

Comment:
That's fair. 
Initially, the purpose was to just capture the magic number 8, so that we could expand our library of known tuples without chasing them. Then I realized I don't want to do `8 -1` in lots of places. 
Basically, `8` is the max size of ValueTuple and `7` is the same thing as zero-based. I'm open to better names, but I don't think we need to introduce more flavors.